src/Http/ResponseTransformer.php

```diff
@@ -162,8 +162,20 @@ public static function toPsr(CakeResponse $response)
         if (!isset($headers['Content-Type'])) {
             $headers['Content-Type'] = $response->type();
         }
-        if ($response->cookie()) {
-            $headers['Set-Cookie'] = static::buildCookieHeader($response->cookie());
+        $cookies = $response->cookie();
+        if ($cookies) {
+            $sessionCookie = session_get_cookie_params();
+            $sessionName = session_name();
+            $cookies[$sessionName] = [
+                'name' => $sessionName,
+                'value' => session_id(),
+                'expire' => $sessionCookie['lifetime'],
+                'path' => $sessionCookie['path'],
+                'secure' => $sessionCookie['secure'],
+                'domain' => $sessionCookie['domain'],
+                'httpOnly' => $sessionCookie['httponly'],
+            ];
```

Comment:
Only if there are other cookies. From what data we have here, we can't determine whether or not the session id, or any of the relevant session cookie parameters have changed, so I erred on the side of sending the session cookie back just in case.